entity/shaders/gaussian_blur.frag

```diff
@@ -40,8 +43,18 @@ void main() {
   for (float i = -v_blur_radius; i <= v_blur_radius; i++) {
     float gaussian = Gaussian(i);
     total_gaussian += gaussian;
-    total += gaussian * SampleWithBorder(v_texture_coords +
-                                         v_blur_direction * i / v_texture_size);
+    total +=
+        gaussian * SampleWithBorder(texture_sampler,
+                                    v_texture_coords +
+                                        v_blur_direction * i / v_texture_size);
   }
-  frag_color = total / total_gaussian;
+
+  vec4 blur_color = total / total_gaussian;
+
+  vec4 src_color = SampleWithBorder(alpha_mask_sampler, v_src_texture_coords);
+  float blur_factor = v_inner_blur_factor * src_color.a +
+                      v_outer_blur_factor * (1.0 - src_color.a);
+
+  frag_color =
+      blur_color * blur_factor + src_color * v_src_factor * src_color.a;
```

Comment:
First, doesn't src_color * src_color.a double premultiply the source color when we have src_factor?

Second, the inner part should always be the original, no additional alpha? The right half of this equation will (ignorig the potential double-premultiply) contribute the full interior color and the left half will contribute some more from the gaussian. That should over-apply the inner color for the case of an inner filter.

Reply:
Good catch, removed. The added `src_color.a` factor was leftover from a previous (failed) attempt at making both passes use the same style option. We definitely don't want it.

> That should over-apply the inner color for the case of an inner filter.

`v_src_factor` is set to zero for `kInner`.

Reply:
Ah, I see what you mean. When the mode is `kSource`, the inner blur should never mix with the source color. I replaced the src_color alpha factors with a strict binary mask instead to get this behavior for strokes/fills.

Reply:
The blur factor is now this:
```cpp
  float blur_factor = v_inner_blur_factor * float(src_color.a > 0) +
                      v_outer_blur_factor * float(src_color.a == 0);
```

Reply:
That looks more like what I was expecting. Still approved...